features/netsocket/NetworkStack.h

```diff
@@ -153,6 +153,15 @@ class NetworkStack: public DNS {
      */
     virtual nsapi_error_t add_dns_server(const SocketAddress &address, const char *interface_name = NULL);
 
+    /** Get domain name server address by index from the list of servers to query.
+     *
+     *  @param index  Index of the DNS server, starts from zero.
+     *  @param address  DNS server host address.
+     *  @param interface_name  Network interface name
+     *  @return         NSAPI_ERROR_OK on success, negative error code on failure.
+     */
+    virtual nsapi_error_t get_dns_server_by_index(uint8_t index, SocketAddress *address, const char *interface_name);
```

Comment:
Add interface_name = NULL also and here.

Reply:
Done